rpc_jobs/standard_job_premerge.yml

```diff
@@ -0,0 +1,169 @@
+    # which contains the scripts which will be
+    # executed.
+    repo_url: "https://github.com/rcbops/sample"
+
+    # This is the whitelist for branch names which
+    # should trigger the job automatically. Each
+    # item is used as a Java regex match.
+    branches:
+      - "master"
+
+    # This is a list of names to designate the image
+    # used when preparing a cloud instance. Each name
+    # must also include the key IMAGE with a value
+    # that corresponds to the appropriate image UUID
+    # or name of the image available in the cloud.
+    image:
+      - xenial:
+          IMAGE: "Ubuntu 16.04 LTS (Xenial Xerus) (PVHVM)"
+
+    # This is a list of names to designate the 'scenario'
+    # which is being executed. This provides the ability
+    # to execute the same scripts with different values
+    # which affect the code path tested.
+    scenario:
+      - "default"
+
+    # This is a list of names to designate the 'action'
+    # which is executed. This provides the ability to
+    # execute different scripts.
+    action:
+      - "lint"
+      - "deploy"
+      - "upgrade"
+
+    # If an alternative FLAVOR is needed, (eg: when the
+    # job intended to be used with onmetal) then one of
+    # the axes (image, scenario, action) should specify
+    # the FLAVOR parameter in a similar way to the IMAGE
+    # parameter.
+
+    jobs:
+      - 'RPC-{name}-{image}-{scenario}-{action}'
+
+- job-template:
+    name: 'RPC-{name}-{image}-{scenario}-{action}'
+    project-type: workflow
+    concurrent: true
+    properties:
+      - build-discarder:
+          num-to-keep: "30"
+      - github:
+          url: "{repo_url}"
+    parameters:
+      - rpc_gating_params
+      - instance_params:
+          IMAGE: "{IMAGE}"
+          FLAVOR: "{FLAVOR}"
+          REGION: "{REGION}"
+      - string:
+          name: STAGES
+          default: >
+            Allocate Resources,
+            Connect Slave,
+            Cleanup,
+            Destroy Slave
+          description: |
+            Pipeline stages to run CSV. Note that this list does not influence execution order.
+            Options:
+              Allocate Resources
+              Connect Slave
+              Cleanup
+              Destroy Slave
+    triggers:
+      - github-pull-request:
+          org-list:
+            - rcbops
+          github-hooks: true
+          trigger-phrase: '.*recheck_cit_all.*|.*recheck_cit_{image}_{scenario}_{action}.*'
+          only-trigger-phrase: false
+          white-list-target-branches: "{branches}"
+          auth-id: "github_account_rpc_jenkins_svc"
+          status-context: 'CIT/{image}_{scenario}_{action}'
+          cancel-builds-on-update: true
+
+    dsl: |
+      library "rpc-gating@${{RPC_GATING_BRANCH}}"
+
+      // Pass details about the job parameters through
+      // to the target environment so that scripts can
+      // use them to adapt behaviour.
+      env.CI_JOB_NAME = "{name}"
```

Comment:
I used RE_ for the hook env vars, maybe make sense to follow?

Reply:
Maybe. I was thinking that ``CI_JOB_`` as a prefix makes the intent clearer, and that maybe our other env vars should change.

Reply:
I used RE because not all RE hooks will be CI related. Some stuff will be to do with project life cycle rather than integration testing.

Reply:
Fair enough. These are CI-specific though.